src/coreclr/nativeaot/Runtime/GCHelpers.cpp

```diff
@@ -132,6 +133,13 @@ COOP_PINVOKE_HELPER(void, RhUnregisterGcCallout, (GcRestrictedCalloutKind eKind,
     RestrictedCallouts::UnregisterGcCallout(eKind, pCallout);
 }
 
+#ifdef FEATURE_OBJCMARSHAL
+COOP_PINVOKE_HELPER(void, RhRegisterObjcCallbacks, (void * pTryGetTaggedMemoryCallback, void * pBeginEndCallback, void * pIsReferencedCallback, void * pTrackedObjectEnteredFinalization))
```

Comment:
We do callbacks into CoreLib a bit differently to support a scenario where a single instance of the runtime (GC et al.) is running multiple independent "managed worlds" with their own `System.Object`. (Consider a scenario where the runtime is built into a single dynamic library and there are multiple other dynamic libraries, all with managed code in it and their own `System.Object` version sharing this singular runtime within a single process).

Look for `AppendExceptionStackFrame` or `IDynamicCastableIsInterfaceImplemented` under src/coreclr/nativeaot for samples. I can help if you have more questions.

The idea is to "grab the function pointer that corresponds to the given `MethodTable`" - which is what `IDynamicCastableIsInterfaceImplemented` does.

Reply:
Unless it's perf critical and we cannot afford an extra call instruction, I'd go with the most straightforward option "Add new class lib functions that return the user-provided function pointers.".

Reply:
Sounds good, I updated the PR with that approached.